zipline/data/minute_bars.py

```diff
@@ -950,3 +957,107 @@ def load_raw_arrays(self, fields, start_dt, end_dt, sids):
                 out *= self._ohlc_inverse
             results.append(out)
         return results
+
+
+class PanelMinuteBarReader(object):
+    """
+    Reader for data passed as Panel.
+
+    DataPanel Structure
+    -------
+    items : Int64Index
+        Asset identifiers.  Must be unique.
+    major_axis : DatetimeIndex
+       Datetimes for data provided by the Panel.  Must be unique.
+    minor_axis : ['open', 'high', 'low', 'close', 'volume']
+       Price attributes.  Must be unique.
+
+    Attributes
+    ----------
+    The table with which this loader interacts contains the following
+    attributes:
+
+    panel : pd.Panel
+        The panel from which to read OHLCV data.
+    first_trading_day : pd.Timestamp
+        The first trading day in the dataset.
+    """
+    @preprocess(panel=call(verify_indices_all_unique))
+    def __init__(self, calendar, panel):
+
+        panel = panel.copy()
+        if 'volume' not in panel.minor_axis:
+            # Fake volume if it does not exist.
+            panel.loc[:, :, 'volume'] = int(1e9)
+
+        self.first_trading_day = pd.datetools.normalize_date(
+            panel.major_axis[0]
+        )
+        self._calendar = calendar
+
+        self.panel = panel
+
+        self._ohlc_inverse = 1. / OHLC_RATIO
+
+    @property
+    def last_available_dt(self):
+        return self.panel.major_axis[-1]
+
+    def load_raw_arrays(self, columns, start_dt, end_dt, assets):
```

Comment:
I think we can vectorize this operation:
```python
return list(self.panel.loc[list(assets), start_dt:end_dt, list(columns)].values.T)
```

This works by slicing the panel down by all the constraints and then asking for the data as an ndarray. This will give us a `(sids, dt, columns)` array. We actually need to get the a `(dt, sid)` per column, so by taking the transpose we get a `(columns, dt, sids)` array. We can then iterate over this to take slices along the column axis to give us `len(columns)` `(dt, sid)` arrays.